fdmi/plugins/m0scheduler

```diff
@@ -17,128 +17,277 @@
 # For any questions about this software or licensing,
 # please email opensource@seagate.com or cortx-questions@seagate.com.
 #
+
+# This is FDMI plugin sample app wrapper that serves as a scheduler
+# for the FDMI records. It gathers that cluster infp using Hare hctl
+# tool, forks m0sched program and waits for records delivered by it.
+# For now it is just a demo that prints FDMI records to console, but
+# later it can perform records de-dup and replication coordination by
+# means of set of the replication python programs that perform actual
+# data move.
+
 import subprocess
 import signal
-from socket import gethostname
 import shlex
 import sys
 import json
-import ast
 import codecs
 
-kv_records = [] #global kv cache
+cluster_info = dict()  # global cluster info
+kv_records = []        # global kv cache
+processes = []         # global list if m0sched processes
+
 
 def str_decode(encoded):
-	if encoded == "0" :
-		return encoded;
-	#return codecs.decode(encoded, 'hex').decode('utf-8')
-	return codecs.decode(encoded, 'hex')
+    if encoded is None:
+        return None
+    if encoded == '0':
+        return encoded
+    # return codecs.decode(encoded, 'hex').decode('utf-8')
+    return codecs.decode(encoded, 'hex')
+
 
 def json_parser(kv_record):
```

Comment:
`s/json_parser/process_fdmi_record/ ` ? It processes the record, json parse, dedup and print

Reply:
Is it the python part that does dedup?

Reply:
Yes here in sample, it does, see line #73 below.

Reply:
agreed